anchore_engine/analyzers/utils.py

```diff
@@ -2047,3 +2048,22 @@ def write_kvfile_fromdict(file, indict):
             thestr = ' '.join([cleank, dict[k], '\n'])
             #thestr = thestr.encode('utf8')
             OFH.write(thestr)
+
+def defaultdict_to_dict(d):
+    if isinstance(d, collections.defaultdict):
+        d = {k: defaultdict_to_dict(v) for k, v in d.items()}
+    return d
+
+def merge_nested_dict(a, b, path=None):
+    if path is None: path = []
```

Comment:
this doesn't read right because it is inlining code with the `:`. Surprised the linter didn't complain about this! 
```suggestion
    if path is None: 
        path = []
```